### PR TITLE
Fix mixin for string_application

### PR DESCRIPTION
**Pull request by jakobod**
This PR fixes the templates for Mixins in the string_application test.

### CODE REVIEW

**Comment by josephnoir** [comment]
Since I'm not entirely sure what the implication is when not using the second parameter for extend (does this result in the type the same type for `Subtype` and `Base`?), would you mind taking a look at this @Neverlord?

File: libcaf_net/test/string_application.cpp [hunk 1]
```diff
@@ -194,12 +197,14 @@ CAF_TEST(receive) {
                   sec::unavailable_or_would_block);
   CAF_MESSAGE("adding both endpoint managers");
   auto mgr1 = make_endpoint_manager(mpx, sys, policy::scribe{sockets.first},
-                                    stream_string_application{sys, buf});
+                                    extend<string_application>::with<
```
Comment (Neverlord) [inline libcaf_net/test/string_application.cpp]: This rather unwieldy type is used twice. I'd give it a type alias to make the code more readable and less repetitive:

```c++
  using application_type = extend<string_application>::with<stream_string_application>;
```

**Comment by Neverlord** [comment]
> would you mind taking a look at this @Neverlord?

Not at all. 🙂

> Since I'm not entirely sure what the implication is when not using the second parameter for extend (does this result in the type the same type for Subtype and Base?)

Some mixins use the second argument for CRTP. In this case, not setting the optional second argument is fine.

File: libcaf_net/test/string_application.cpp (hunk 1 above)
Comment (jakobod) [reply libcaf_net/test/string_application.cpp]: Thanks for the tip @Neverlord.

**Pull request merged**